### DISCUSSION TITLE
Supported CHANGELOG.md formats?

### DISCUSSION THREAD

**Discussion by nielsbasjes** [renovatebot/renovate#10528] [Request Help]
Hi,

I have renovate running (internally) and it correctly downloads and shows the changes related to the version jump in the created merge request.
So apparently I have configured everything correctly and it understands the formatting I have used in my `CHANGELOG.md`.
I consider this an `accidental success` ... I accidentally formatted my CHANGELOG.md in a way that Renovate understands.

What are the supported formats for the CHANGELOG ?
I have not been able to find any documentation on that.

Niels

**Comment by HonkingGoose** [comment]
> What are the supported formats for the CHANGELOG ?
> I have not been able to find any documentation on that.

I couldn't find any docs on supported changelogs as well... 😄 

We should put a small section about changelog support in the docs. I'm willing to make a draft for the documentation, once the Renovate code wizards tell me what we support. 😉

**Reply by nielsbasjes** [reply]
How about the practical approach: Just show some examples of well known projects that work well in the documentation?

**Comment by rarkins** [comment]
It's pretty complicated but there's two main parts of importance:

The first thing it does is try to work out which heading level indicates releases, by seeing which level has the most: https://github.com/renovatebot/renovate/blob/6b252ba23d6b041724e42b22e02c84b8df775da0/lib/workers/pr/changelog/release-notes.ts#L261-L300

Then it tries to match headings against releases using a few combinations: https://github.com/renovatebot/renovate/blob/6b252ba23d6b041724e42b22e02c84b8df775da0/lib/workers/pr/changelog/release-notes.ts#L108-L114

**Comment by herndlm** [comment]
> Hi,
> 
> I have renovate running (internally) and it correctly downloads and shows the changes related to the version jump in the created merge request.
> So apparently I have configured everything correctly and it understands the formatting I have used in my `CHANGELOG.md`.
> I consider this an `accidental success` ... I accidentally formatted my CHANGELOG.md in a way that Renovate understands.
> 
> What are the supported formats for the CHANGELOG ?
> I have not been able to find any documentation on that.
> 
> Niels

I'm now curious too. Are you generating your changelogs with some tooling? It would be great if we can figure out the format, document that, add some examples as you mentioned and maybe some tool links for generation (e.g. based on conventional commits). I'll look at the code later too.

**Reply by HonkingGoose** [reply]
Let me plug a project that I'm helping to improve, it's called Octoclairvoyant. The idea is that it fetches the GitHub changelogs and puts them in a single view. You might like to take a look at the website/code for inspiration? 😉 

Link to Octoclairvoyant repo on GitHub: https://github.com/Belco90/octoclairvoyant

Link to live app/website: https://octoclairvoyant.vercel.app/about
